src/Microsoft.AspNetCore.Mvc.Razor/Compilation/ViewsFeatureProvider.cs

```diff
@@ -91,7 +94,7 @@ protected virtual IReadOnlyList<RazorCompiledItem> LoadItems(AssemblyPart assemb
                 throw new ArgumentNullException(nameof(assemblyPart));
             }
 
-            var viewAssembly = GetViewAssembly(assemblyPart);
+            var viewAssembly = assemblyPart.Assembly;
```

Comment:
The changes to this method are a subtle breaking change. We discussed detecting if this type has been subclassed and providing the legacy behavior if that's the case. 

I'm not sure if it matters - we're providing a different list of application parts, and the expectations are different now. @Eilon thoughts?

Reply:
Isn't this method an API breaking change? It was a public API and now it's private?

Reply:
As far as returning a different set of assemblies, assuming you're not using any of the new opt-in behaviors (because you wouldn't be), what exactly would be different for any existing app?

Reply:
We are already explicitly ordering the list of application parts, so the ViewDescriptors in ViewsFeature are already getting re-arranged, that said, we do a fair amount of work to ensure we are non-breaking.
* If you have a custom ViewsFeatureProvider, we will detect the override at Runtime and invoke the old behavior.
* We will also put the list of "new parts" (precompiled view parts) after all the parts in your app, so the result is that any descriptor that happens as part of any of those parts, will happen later in the list, thus preserving any previous overriding behavior you might have in your views feature.

What I'm trying to say, is that I do believe that these set of changes might alter the intermediate visible state but that the final observable state should be the same, and that we are already changing that intermediate state (the ViewsFeature) based on the fact that we decided to order the list of application parts. (Which also potentially impacts any other FeatureProvider)

Reply:
The change of the contents of the list seems fairly innocuous to me.

But, unless this API is new in 2.1.0, isn't this removing a public API?

Reply:
This api was added by me like a week ago 😆 - it's new in 2.1